docs/contributing/code-review.md

```diff
@@ -0,0 +1,73 @@
+# Code review guidelines
+
+This document codifies our guidelines for doing code review
+and merging pull requests.
+
+> A Foolish Consistency is the Hobgoblin of Little Minds
+> - PEP 8
+
+## Terraform Changes
+
+[Terraform](https://www.terraform.io/) lets us treat our infrastructure
+as code, so we can use regular code quality practices (code review,
+linters, extensive documentation, etc) to keep the quality of our
+infrastructure high. However, it comes with a few caveats:
+
+1. Locally and iteratively testing terraform is impossible - as we
+   develop it, the infrastructure is deployed / modified / destroyed
+   as we go.
+2. Behavior of a bit of terraform code differs based on the current
+   state of the infrastructure (as represented in the
+   [state file](https://www.terraform.io/docs/language/state/index.html)).
+3. While `terraform plan` can tell us what terraform *thinks* it is
+   going to do, it might not always succeed. Permission errors or
+   organizational restrictions can cause it to fail. Timeouts can happen.
+   Runtime errors pop up.
```

Comment:
I like this section explaining the associated complexities!

Reply:
yay!